src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -979,3 +975,34 @@ ColumnType(Oid relationId, char *columnName)
 
 	return columnType;
 }
+
+
+/*
+ * Check that the current replication factor setting is compatible with the
+ * replication model of relationId, if valid. If InvalidOid, check that the
+ * global replication model setting instead. Errors out if an invalid state
+ * is detected.
+ */
+void
+EnsureReplicationSettings(Oid relationId)
+{
+	char replicationModel = (char) ReplicationModel;
+	char *msgSuffix = "the streaming replication model";
+	char *extraHint = " or setting \"citus.replication_model\" to \"statement\"";
+
+	if (relationId != InvalidOid)
+	{
+		replicationModel = TableReplicationModel(relationId);
+		msgSuffix = "tables which use streaming replication";
```

Comment:
I don't know if that adds much to the user other than a technical-sounding word, but I'm mulling it over and might switch.